numpy/_typing/_ufunc.pyi

```diff
@@ -30,10 +31,10 @@ _2Tuple = tuple[_T, _T]
 _3Tuple = tuple[_T, _T, _T]
 _4Tuple = tuple[_T, _T, _T, _T]
 
-_NTypes = TypeVar("_NTypes", bound=int)
-_IDType = TypeVar("_IDType", bound=Any)
-_NameType = TypeVar("_NameType", bound=str)
-_Signature = TypeVar("_Signature", bound=str)
+_NTypes = TypeVar("_NTypes", bound=int, covariant=True)
+_IDType = TypeVar("_IDType", bound=Any, covariant=True)
+_NameType = TypeVar("_NameType", bound=str, covariant=True)
+_Signature = TypeVar("_Signature", bound=str, covariant=True)
```

Comment:
It's not entirely clear to me whether this change is needed or why it's an improvement. The test suite passes either way. Why does it matter?

Reply:
If you'd have used `TypeVar(_, infer_variance=True)` here, then typecheckers would infer this as covariant as well, as it's only used as return type or (readonly) attribute type.

In this case, this means that any function or type with e.g. `x: _UFunc_Nin1_Nout1[int, str, None]` as input parameter, *only* allows e.g. `_UFunc_Nin1_Nout1[Literal[4], Literal['spam'], None]` if the first two type params of `_UFunc_Nin1_Nout1` are covariant. 
Otherwise (if invariant), the type argument types must be exactly the same as that of their respective type parameters, i.e. only `_UFunc_Nin1_Nout1[str, int, None]` is accepted as input.

Reply:
Makes sense, thanks.